src/http/server.rs

```diff
@@ -799,6 +803,108 @@ impl EspHttpRawConnection<'_> {
 
         Ok(())
     }
+
+    /// Retrieves the source IPv4 of the request.
+    ///
+    /// The IPv4 is retrieved using the underlying session socket.
+    #[cfg(esp_idf_lwip_ipv4)]
+    pub fn source_ip4(&self) -> Result<Ipv4Addr, EspError> {
+        unsafe {
+            let sockfd = httpd_req_to_sockfd(self.handle());
+
+            if sockfd == -1 {
+                return Err(EspError::from(-1).unwrap());
+            }
+
+            let mut addr = sockaddr_in {
+                sin_len: core::mem::size_of::<sockaddr_in>() as u8,
+                sin_family: AF_INET as u8,
+                ..Default::default()
+            };
+
+            let mut addr_len = core::mem::size_of::<sockaddr_in>() as socklen_t;
+
+            esp!(lwip_getpeername(
+                sockfd,
+                &mut addr as *mut _ as *mut sockaddr,
+                &mut addr_len,
+            ))?;
+
+            //"255.255.255.255\0" = 16 bytes
```

Comment:
The whole `sin_addr`-to-char-then-to-`Ipv4Addr` equilibristic is unnecessary.

`sin_addr.s_addr` is of type `uint32_t` = `u32` and represents the Ipv4 address either in big or little endian (can't remember which one it was) but you can easily figure it out.

If you need to get the bytes of it, either do `sin_addr.s_addr.,to_le_bytes()` ot `sin_addr.s_addr.to_be_bytes()`.